play-ahc-ws-standalone/src/main/java/play/libs/ws/ahc/StandaloneAhcWSRequest.java

```diff
@@ -221,7 +223,16 @@ public StandaloneAhcWSRequest setRequestTimeout(long timeout) {
         if (timeout < -1 || timeout > Integer.MAX_VALUE) {
             throw new IllegalArgumentException("Timeout must be between -1 and " + Integer.MAX_VALUE + " inclusive");
         }
-        this.timeout = (int) timeout;
+        this.timeout = Duration.of(timeout, ChronoUnit.MILLIS);
+        return this;
+    }
+
+    @Override
+    public StandaloneAhcWSRequest setRequestTimeout(Duration timeout) {
```

Comment:
While the max value for the `setRequestTimeout(long)` is `Integer.MAX_VALUE`, this version will accepts `Duration`s with values bigger than that. This won't be a problem though.